Build failure: python311Packages.nbmake

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `python311Packages.nbmake` on master (ba2e291ee041c6b93448a47191ee32bfc3f08592) a few times

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

This build log is from nixos-unstable, didn't bother to reproduce it on master. It is truncated as the `nix log ...` command didn't output anything.

<summary>Build Log</summary>

```
error: builder for '/nix/store/dkgjc7l7ysi5dmhv4a7rjf6rlwbcfgc8-python3.11-nbmake-1.5.4.drv' failed with exit code 1;                                                                                             
       last 25 log lines:                                                                                                                                                                                         
       > KeyboardInterrupt:                                                                                                                                                                                       
       > ------------------------------ Captured log call -------------------------------                                                                                                                         
       > ERROR    traitlets:client.py:568 Error occurred while starting new kernel client for kernel b73c6d04-b562-4bc2-8529-1a083a596efa: Kernel didn't respond in 60 seconds                                    
       >                                                                                                                                                                                                          
       > Learn more about nbmake at https://github.com/treebeardtech/nbmake                                                                                                                                       
       >                                                                                                                                                                                                          
       > =========================== short test summary info ============================                                                                                                                         
       > FAILED 10.ipynb::                                                                                                                                                                                        
       > =================== 1 failed, 19 passed in 74.41s (0:01:14) ====================                                                                                                                         
       > ----------------------------- Captured stderr call -----------------------------                                                                                                                         
       > Exception ignored in atexit callback: <bound method run_sync.<locals>.wrapped of <nbclient.client.NotebookClient object at 0xfffff4135990>>                                                              
       > Traceback (most recent call last):                                                                                                                                                                       
       >   File "/nix/store/qvs9sja4rzf1bdgyna6l6xmf9njdzj3i-python3.11-jupyter-core-5.7.2/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped                                     
       >     return loop.run_until_complete(inner)                                                                                                                                                                
       >            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                
       >   File "/nix/store/cljp9nclqkbk08n6i1yk7ii4vv96pl68-python3-3.11.11/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete                                                              
       >     return future.result()                                                                                                                                                                               
       >            ^^^^^^^^^^^^^^^                                                                                                                                                                               
       >   File "/nix/store/k9crdvfhd89jh7422rv74bgxpnjmak5v-python3.11-nbclient-0.10.0/lib/python3.11/site-packages/nbclient/client.py", line 504, in _async_cleanup_kernel                                      
       >     assert self.km is not None                                                                                                                                                                           
       >            ^^^^^^^^^^^^^^^^^^^                                                                                                                                                                           
       > AssertionError:                                                                                                                                                                                          
       > =========================== short test summary info ============================                                                                                                                         
       > FAILED tests/test_pytest_plugin.py::test_when_parallel_passing_nbs_then_ok - assert <ExitCode.TESTS_FAILED: 1> == <ExitCode.OK: 0>                                                                       
       > =================== 1 failed, 32 passed in 118.60s (0:01:58) ===================                                                                                                                         
       For full logs, run 'nix log /nix/store/dkgjc7l7ysi5dmhv4a7rjf6rlwbcfgc8-python3.11-nbmake-1.5.4.drv'.                                                                                                      
```

</details>


The build either:
- succeeds (most of the time, finishes in 2 minutes)
- fails (see build log)
- hangs (fixed by https://github.com/NixOS/nixpkgs/pull/369950)

Screenshot of hanging at tests for 7 hours after the 4th attempt:

![Screenshot_20241231-204043_Termux](https://github.com/user-attachments/assets/04a016e3-eb01-46b9-bd6c-b66c14484817)



## Additional context

I ran into this when trying to build `helix`.

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 6.11.5, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: aarch64-linux
Additional system types: 
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/gep/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/gep/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/gep/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/gep/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/2si9cm7zi0pcx0xj3wh98h039njyv3nk-lix-2.91.1/share`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@GaetanLepage 

Related:
- https://github.com/NixOS/nixpkgs/pull/367720
- https://github.com/NixOS/nixpkgs/pull/359867 @FliegendeWurst 



---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

> Try ` pytestFlagsArray = [ "--maxprocesses=1" ];`

@FliegendeWurst 

This fixed the timeout issue, but a test still fails sometimes (1 in 100). The relevant error message seems to be
```
ERROR    traitlets:client.py:568 Error occurred while starting new kernel client for kernel cdda69d0-1f5f-4900-9dbb-61105a780aed: Kernel died before replying to kernel_info
```

Log of 100 builds including the 1 failing:

```
gep@raspi-doboz nixpkgs ❯ nix build .#python311Packages.nbmake
warning: Git tree '/home/gep/nixpkgs' is dirty
gep@raspi-doboz nixpkgs ❯ for i in {1..100}; do date; nix build .#python311Packages.nbmake --rebuild && echo success || echo fail; done
Wed Jan  1 02:55:18 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
[1/0/1 built] building python3.11-nbmake-1.5.4 (pytestCheckPhase): 1 wo
success
Wed Jan  1 02:58:27 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:01:37 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:04:48 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:08:00 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:11:11 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:14:24 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:17:34 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:20:43 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:23:54 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:27:05 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:30:16 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:33:27 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:36:39 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:39:49 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:42:59 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:46:07 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:49:18 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:52:28 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:55:37 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 03:58:47 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:02:10 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:05:19 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:08:30 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:11:41 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:14:52 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:18:02 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:21:13 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:24:23 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:27:50 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:31:05 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:34:14 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:37:23 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:40:34 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:43:46 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:46:56 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:50:08 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:53:20 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:56:30 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 04:59:42 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:02:54 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:06:06 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:09:16 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:12:26 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:15:35 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:18:44 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:21:54 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:25:09 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:28:24 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:31:38 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:34:50 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:38:01 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:41:13 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:44:24 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:47:34 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:50:44 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:53:53 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 05:57:01 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:00:11 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:03:21 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:06:32 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:09:41 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:12:50 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
error: builder for '/nix/store/gw4ssl52yc89c1kky4l4mhiglfbvb4ys-python3.11-nbmake-1.5.4.drv' failed with exit code 1;
       last 25 log lines:
       > zmq.error.ZMQError: Address already in use (addr='tcp://127.0.0.1:48717')
       > ------------------------------ Captured log call -------------------------------
       > ERROR    traitlets:client.py:568 Error occurred while starting new kernel client for kernel cdda69d0-1f5f-4900-9dbb-61105a780aed: Kernel died before replying to kernel_info
       >
       > Learn more about nbmake at https://github.com/treebeardtech/nbmake
       >
       > =========================== short test summary info ============================
       > FAILED 17.ipynb::
       > ======================== 1 failed, 19 passed in 31.12s =========================
       > ----------------------------- Captured stderr call -----------------------------
       > Exception ignored in atexit callback: <bound method run_sync.<locals>.wrapped of <nbclient.client.NotebookClient object at 0xffffef31ccd0>>
       > Traceback (most recent call last):
       >   File "/nix/store/qvs9sja4rzf1bdgyna6l6xmf9njdzj3i-python3.11-jupyter-core-5.7.2/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
       >     return loop.run_until_complete(inner)
       >            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/cljp9nclqkbk08n6i1yk7ii4vv96pl68-python3-3.11.11/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
       >     return future.result()
       >            ^^^^^^^^^^^^^^^
       >   File "/nix/store/k9crdvfhd89jh7422rv74bgxpnjmak5v-python3.11-nbclient-0.10.0/lib/python3.11/site-packages/nbclient/client.py", line 504, in _async_cleanup_kernel
       >     assert self.km is not None
       >            ^^^^^^^^^^^^^^^^^^^
       > AssertionError:
       > =========================== short test summary info ============================
       > FAILED tests/test_pytest_plugin.py::test_when_parallel_passing_nbs_then_ok - assert <ExitCode.TESTS_FAILED: 1> == <ExitCode.OK: 0>
       > =================== 1 failed, 32 passed in 162.24s (0:02:42) ===================
       For full logs, run 'nix log /nix/store/gw4ssl52yc89c1kky4l4mhiglfbvb4ys-python3.11-nbmake-1.5.4.drv'.
fail
Wed Jan  1 06:16:01 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:19:11 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:22:22 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:25:30 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:28:41 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:31:50 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:35:01 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:38:11 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:41:23 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:44:34 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:47:44 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:50:56 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:54:07 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 06:57:16 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:00:27 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:03:38 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:06:49 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:09:59 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:13:11 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:16:23 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:19:31 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:22:42 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:25:50 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:29:02 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:32:13 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:35:22 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:38:33 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:41:42 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:44:54 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:48:03 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:51:11 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:54:21 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 07:57:32 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 08:00:42 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 08:03:53 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 08:07:04 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
Wed Jan  1 08:10:15 AM CET 2025
warning: Git tree '/home/gep/nixpkgs' is dirty
success
gep@raspi-doboz nixpkgs ❯
```

--- a/pkgs/development/python-modules/nbmake/default.nix
+++ b/pkgs/development/python-modules/nbmake/default.nix
@@ -13,7 +13,6 @@
   pygments,
 
   # tests
-  pytest-xdist,
   pytestCheckHook,
   typing-extensions,
 }:
@@ -45,12 +44,7 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "nbmake" ];
 
-  # tests are prone to race conditions under high parallelism
-  # https://github.com/treebeardtech/nbmake/issues/129
-  pytestFlagsArray = [ "--maxprocesses=4" ];
-
   nativeCheckInputs = [
-    pytest-xdist
     pytestCheckHook
     typing-extensions
   ];
@@ -58,6 +52,13 @@ buildPythonPackage rec {
   preCheck = ''
     export HOME=$(mktemp -d)
   '';
+
+  disabledTests = [
+    # depends on pytest-xdist that is not added, as
+    # tests are prone to race conditions under parallelism, they sometimes hang indefinitely
+    # https://github.com/treebeardtech/nbmake/issues/129
+    "test_when_parallel_passing_nbs_then_ok"
+  ];
 
   __darwinAllowLocalNetworking = true;
 

